hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/token/impl/TokenStore.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.service.token.impl;
+
+import com.hedera.node.app.spi.state.State;
+import com.hedera.node.app.spi.state.States;
+import com.hedera.services.legacy.core.jproto.JKey;
+import com.hedera.services.state.merkle.MerkleToken;
+import com.hedera.services.state.submerkle.EntityId;
+import com.hedera.services.state.submerkle.FcCustomFee;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.TokenID;
+
+import javax.annotation.Nonnull;
+import java.util.Optional;
+
+/**
+ * Provides methods for interacting with the underlying data storage mechanisms for working with
+ * Tokens.
+ *
+ * <p>This class is not exported from the module. It is an internal implementation detail.
+ */
+public class TokenStore {
+	/** The underlying data storage class that holds the token data. */
+	private final State<Long, MerkleToken> tokenState;
+
+	/**
+	 * Create a new {@link TokenStore} instance.
+	 *
+	 * @param states The state to use.
+	 */
+	public TokenStore(@Nonnull final States states) {
+		this.tokenState = states.get("TOKENS");
+	}
+
+
+	public record TokenMetadata(Optional<JKey> adminKey,
+								Optional<JKey> kycKey,
+								Optional<JKey> wipeKey,
+								Optional<JKey> freezeKey,
+								Optional<JKey> supplyKey,
+								Optional<JKey> feeScheduleKey,
+								Optional<JKey> pauseKey,
+								boolean hasRoyaltyWithFallback,
+								EntityId treasury) {}
+
+	public record TokenMetaOrReason(TokenMetadata metadata, ResponseCodeEnum failureReason) {
```

Comment:
Maybe `TokenResult`, `TokenOperationResult`?

Reply:
Used `OrReason` following the same naming as `KeyorLookupFailureReason` in `AccountStore`